mqtt/mqtt-broker/src/broker.rs

```diff
@@ -29,9 +34,64 @@ macro_rules! try_send {
     }};
 }
 
-pub struct Broker<Z> {
+pub struct BrokerRunner<Z> {
```

Comment:
can you please clarify why do we need this separate struct?

Reply:
Broker used to have 2 "parts" messages receiving and messages processing logic. method `run` deconstructs broker instance so we could different functions can own different parts of the broker. So the separation I made is somewhat intuitive IMO.
The problem I ran into: `Receiver::recv()` takes `&mut self`, then I make `pin_mut` its future and pass ownership to `future::select()`. So it makes it impossible to have another borrow of mutable reference to `self` when I need to process a message which also requires mutable reference to `self`. If I split messages receiving and messages processing between different instance there are no such problem.

Reply:
I see. The second part make sense. Nit naming suggestion `BrokerLoop`?

Reply:
`BrokerRunner` has `run` method that's why it is runner :)